02_bubble/CameronSchafer/bubble.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	// requirement was to sort a copy of the array.
+	t := make([]int, len(s))
+	copy(t, s)
+	check := false
+
+	for !check {
+		tempCheck := true
+		// sort once and then check
+		for i := 0; i < len(t); i++ {
+			// check if in last slice
+			if i+1 < len(t) {
+				if t[i] > t[i+1] {
+					// swap the elems around
+					t[i], t[i+1] = t[i+1], t[i]
+					tempCheck = false
+				}
+			}
+		}
+		check = tempCheck
+	}
+	return t
+}
+
+func insertion(s []int) []int {
+	// loop through the given slice
+	for i := 0; i < len(s); i++ {
+		a, firstPart, secondPart := s[i], s[:i], s[i+1:]
+
+		for j := len(firstPart) - 1; j >= 0; j-- {
+			if a < firstPart[j] {
+				newA, b := a, firstPart[j]
+				// create the new slice and reset the parent loop
+				s, i = append(append(firstPart[:j], newA), append([]int{b}, secondPart...)...), 0
+				break
+			}
+		}
+	}
+	return s
+}
+
+// accidentally created an O(n2) complexity instead of O(n log n) complexity
+// merge sort: O(n2) sorting algorithm
+func merge(s []int) []int {
+	sBox := [][]int{}
+	for a := range s {
+		sBox = append(sBox, []int{s[a]})
```

Comment:
use `make` if you know size of the slice to be generated (more runtime efficient).

Reply:
removed from code

Reply:
will keep in mind for future code 👍